File: go-algorithm/leetcode/problem231.go
package main

// 231. Power of Two
// https://leetcode.com/problems/power-of-two/

func isPowerOfTwo(n int) bool {
	if n <= 0 {
		return false
	}
	if n == 1 {
		return true
	}
	for n > 1 {
		if n%2 != 0 {
			return false
		}
		n /= 2
	}
	return true
}

File: java-algo/src/main/java/com/leetcode/Problem231.java
package com.leetcode;

public class Problem231 {
    public boolean isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
        if (n == 1) {
            return true;
        }
        while (n > 1) {
            if (n % 2 != 0) {
                return false;
            }
            n >>= 1;
        }
        return true;
    }
}
